source/common/upstream/load_balancer_impl.cc

```diff
@@ -747,9 +747,8 @@ const HostVector& ZoneAwareLoadBalancerBase::hostSourceToHosts(HostsSource hosts
     return host_set.healthyHostsPerLocality().get()[hosts_source.locality_index_];
   case HostsSource::SourceType::LocalityDegradedHosts:
     return host_set.degradedHostsPerLocality().get()[hosts_source.locality_index_];
-  default:
-    NOT_REACHED_GCOVR_EXCL_LINE;
   }
+  PANIC_DUE_TO_CORRUPT_ENUM;
```

Comment:
Is this one arguably worth fixing for full coverage or filing a bug for? Would be good to clean these up, even if we just tag them all with a tracking bug and give this to someone who is looking for a starter bug.

Reply:
I guess I'm thinking of this like the audit macros we had for exceptions. We have some places we might genuinely (?) want this in the long term, but we should flag the places we're not sure and let folks go and clean this up on best effort basis. This makes the big sweep you're about to do very mechanical.

Reply:
yeah I agree that's safer. https://github.com/envoyproxy/envoy/issues/19229